File: result/translate/starchat/python2java/java/ARRAY_RANGE_QUERIES_ELEMENTS_FREQUENCY_VALUE.java
public static int f_gold ( int start, int end, int arr [ ] ) {
        HashMap < Integer, Integer > frequency = new HashMap < > ( );
        for ( int i = start ; i <= end ; i ++ ) {
            if ( frequency.containsKey ( arr [ i ] ) ) {
                frequency.put ( arr [ i ], frequency.get ( arr [ i ] ) + 1 );
            } else {
                frequency.put ( arr [ i ], 1 );
            }
        }
        int count = 0;
        for ( int x : frequency.keySet ( ) ) {
            if ( frequency.get ( x ) == frequency.get ( x ).intValue ( ) ) {
                count += 1;
            }
        }
        return count;
    }

File: result/translate/chatgpt/java2cpp/c++/ARRAY_RANGE_QUERIES_ELEMENTS_FREQUENCY_VALUE.cpp
int f_gold(int start, int end, int arr[]) {
    std::unordered_map<int, int> mp;
    for(int i = start; i <= end; i++) {
        mp[arr[i]] = mp[arr[i]] == 0 ? 1 : mp[arr[i]] + 1;
    }

    int count = 0;
    for(auto entry : mp) {
        if(entry.first == entry.second) {
            count++;
        }
    }

    return count;
}